docs/quick-start.md

```diff
@@ -0,0 +1,173 @@
+Download the certificate signing request(CSR) for your leaf certificate
+```bash
+CSR=$(az keyvault certificate pending show --vault-name ${AKV_NAME} --name ${KEY_NAME} --query 'csr' -o tsv)
+CSR_PATH=${KEY_NAME}.csr
+printf -- "-----BEGIN CERTIFICATE REQUEST-----\n%s\n-----END CERTIFICATE REQUEST-----\n" $CSR > ${CSR_PATH}
+```
+
+Create a config file for openssl to sign the laef certificate
+```bash
+cat <<EOF > ./ext.cnf
+[ v3_ca ]
+keyUsage = critical,digitalSignature
+extendedKeyUsage = codeSigning
+EOF
+```
+
+Sign and merge the certificate chain
+```bash
+# sign your leaf certificate by using the CA previously created
+SIGNED_CERT_PATH=${KEY_NAME}.crt
+openssl x509 -CA ca.crt -CAkey ca.key -days 365 -req -in ${CSR_PATH} -set_serial 02 -out ${SIGNED_CERT_PATH} -extensions v3_ca -extfile ./ext.cnf
+
+# merge the certificate chain
+CERTCHAIN_PATH=${KEY_NAME}-chain.pem
+cat ${SIGNED_CERT_PATH} ca.crt > ${CERTCHAIN_PATH}
+```
+
+## Upload the leaf certificate to Azure KeyVault
+```bash
+az keyvault certificate pending merge --vault-name ${AKV_NAME} --name ${KEY_NAME} --file ${CERTCHAIN_PATH}
+```
+
+## Sign with notation
+
+Login to the ACR
+```bash
+export NOTATION_USERNAME="00000000-0000-0000-0000-000000000000"
+export NOTATION_PASSWORD=$(az acr login --name ${REGISTRY} --expose-token --output tsv --query accessToken)
+```
+
+Add key by using azure-kv plugin
+```bash
+KEY_ID=$(az keyvault certificate show -n ${KEY_NAME} --vault-name ${AKV_NAME} --query 'kid' -o tsv)
+notation key add --name ${KEY_NAME} --plugin azure-kv --id ${KEY_ID}
+notation key ls
+```
+
+Choose an authorize mode to visit your keyvault from the plugin
+1. Authorize by Managed Identity(by default)
+    ```bash
+    export AKV_AUTH_METHOD="AKV_AUTH_FROM_MI"
+    ```
+
+    Make sure you have at least granted secret-get, certificate-get, key-sign permissions to your resources.
+
+    Below script is a demo to check and set your azure vm's access policy to your keyvault. For more details, please check https://docs.microsoft.com/en-us/azure/key-vault/general/assign-access-policy?tabs=azure-portal 
+    ```bash
+    # get your azure vm's principalId
+    az vm list --query "[?name=='${VM_NAME}'].identity"
+
+    # get your keyvault's access policy for your azure vm, $OID should be the principalId which we get from the above command
+    az keyvault show --name ${AKV_NAME} --query "properties.accessPolicies[].{objectId:objectId,permissions:permissions}[?contains(objectId,'${OID}')]"
+
+    # if policy doesn't meet the sign requirements, set the access policy.
+    az keyvault set-policy --name ${AKV_NAME} --object-id ${OID} --secret-permissions get --key-permissions sign --certificate-permissions get
+    ```
+2. Authorize by Azure CLI 2.0
+    ```bash
+    export AKV_AUTH_METHOD="AKV_AUTH_FROM_CLI"
+    ```
+    Login to your azure account by Azure CLI
+    ```bash
+    az login
+    az account set --subscription ${subscriptionID}
+    ```
+
+Sign the image
```

Comment:
It is cumbersome to generate keys before signing.

Reply:
+1, we need to discuss a simplified way for configuring the leaf cert and then update this guide.